app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx

```diff
@@ -39,8 +81,20 @@ const cancelButtonString = 'Abort merge'
  */
 export class MergeConflictsDialog extends React.Component<
   IMergeConflictsDialogProps,
-  {}
+  IMergeConflictsDialogState
 > {
+  public constructor(props: IMergeConflictsDialogProps) {
+    super(props)
+
+    this.state = { foundExternalEditor: null }
+  }
+
+  public async componentDidMount() {
```

Comment:
blocking:
what do you think about doing this lookup in `app.tsx` and `appStore`? i'd like to avoid introducing state here as a local callback/re-render mechanism. i think routing this through `dispatcher`, `appStore`, resulting in a `setState` at the app level (which would cause this to rerender) would keep this more global logic out of this component. it would also enable us to rerender this component if in the future something changes the editor state from the background without any additional logic.

Reply:
The biggest headache I had here was that there wasn't an ideal spot to perform this check aside from pushing it all the way back around.

Some other things I considered:

 - `App.currentPopupContent` - the component that shows the dialog - doesn't support doing any async work there
 - pushing this further out to the code that wants to open `PopupType.MergeConflicts` meant having to make changes in four different places
 - I wanted to avoid adding more global state until I could better organize the state to group preferences (which is what `selectedExternalEditor` is currently)

I'll try this pushing this through `dispatcher` and see how it feels after addressing the conflicts.

Reply:
Check out 4ea8718 and 28e137e and see if that helps